Add ::dotWrite method

jodiedunlop opened the pull request:
Added new method for mutating arrays with depth based keys:
`ArrayUtility::dotWrite($array, $key, $value)`

Added new method for testing existence even when the value is null:
`ArrayUtility::dotExists($array, $key)`

Included:
- Tests
- Updated README

jodiedunlop committed:
Add ::dotWrite method

jodiedunlop committed:
Add ::dotExists method

ababkov commented on src/ArrayUtility.php:
```diff
@@ -119,13 +119,72 @@ public static function dotRead(array $array = null, $key, $default_value = null)
+     * @param mixed  $value    Value to write
+     * @return bool  Returns true if write was successful
+     */
+    public static function dotWrite(array &$array, $writeKey, $value = null)
```
Should be immuteable - how come array ref is getting passed in?

jodiedunlop replied:
Should it be immutable? Felt like I wanted to mutate the array without causing a copy op.

ababkov replied:
All the others are immutable

jodiedunlop replied:
Added ::dotMutate, and changed ::dotWrite

ababkov commented on src/ArrayUtility.php:
```diff
@@ -119,13 +119,72 @@ public static function dotRead(array $array = null, $key, $default_value = null)
+    }
+
+    /**
+     * Read a value from a nested array using a single string
```
Probably needs to be updated.

jodiedunlop replied:
Updated

jodiedunlop committed:
Change ::dotWrite to be immutable, add ::dotMutate

- Added ::dotMutate which mutates the array given (by ref)
- Changed ::dotWrite which returns a mutated copy of the array
